src/drivers/common/memory.c

```diff
@@ -100,4 +98,15 @@ static int periph_memory_init(void) {
 	return 0;
 }
 
+void periph_description(struct _segment **buff, int *size) {
+
+	for (int i = 0; i < MAX_SEGMENTS; i++) {
+		if (!_segments[i].start) {
+			*size = i;
```

Comment:
This size is constant and can be obtained with `ARRAY_SPREAD_SIZE(__periph_mem_registry)`. Also it can be just a return value of this function, so no need in the second argument.

Reply:
Fixed